.github/workflows/e2e-test-train-api.yaml

```diff
@@ -0,0 +1,61 @@
+name: E2E Test with train API
+on:
+  - pull_request
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  e2e-test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        kubernetes-version: ["v1.31.4"]
```

Comment:
Just to save compute resources, I think for now we can just run this test on a single k8s version, since we run the rests E2E tests on the all versions.
WDYT @Electronic-Waste ?

Reply:
Yeah, I agree. Maybe we can select one k8s version from this list:)

Reply:
OK, let me change the version to `v1.30.6`. And we can update it if needed in the future.

Reply:
given that we support 1.28-1.31, I would suggest that we run our integration tests on 1.29, 1.30, 1.31, we can update it in the following PR.
For the `train` API tests, I think running it on 1.31 should be sufficient.

Reply:
Sounds good. So I think we will still keep the `v1.31.4` version.